build.ps1

```diff
@@ -254,17 +254,21 @@ function Build-WiXProject()
 {
   [CmdletBinding(PositionalBinding = $false)]
   param(
-    [Parameter(Position = 0)]
+    [Parameter(Position = 0, Mandatory = $true)]
     [string]$FileName,
+    [Parameter(Mandatory = $true)]
+    [hashtable]$Arch,
     [hashtable]$Properties = @{}
   )
 
   $Name = $FileName.Split('.')[0]
+  $ArchName = $Arch.VSName
 
   $Properties = $Properties.Clone()
+  TryAdd-KeyValue $Properties ProductArchitecture $ArchName
   TryAdd-KeyValue $Properties RunWixToolsOutOfProc true
-  TryAdd-KeyValue $Properties OutputPath $BinaryCache\msi\
-  TryAdd-KeyValue $Properties IntermediateOutputPath BinaryCache\$Name\
+  TryAdd-KeyValue $Properties OutputPath $BinaryCache\msi\$ArchName\
+  TryAdd-KeyValue $Properties IntermediateOutputPath BinaryCache\$Name\$ArchName\
```

Comment:
Alternative output path suggestions are welcome

Reply:
🤷‍♂️ Seems like a reasonable thing for now.  we could also flip the two if we wanted, so `$BinaryCache\$ArchName\msi`.  That would isolate it based on the architecture.  I don't feel too strongly about this, its internals and we can restructure this as/when needed.